modules/common/secret/types.go

```diff
@@ -0,0 +1,27 @@
+/*
+Copyright 2024 Red Hat
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package secret
+
+// PasswordSelector
+type PasswordSelector struct {
+	// +kubebuilder:validation:Optional
```

Comment:
the defaulting kubebuilder annotation can not be used, like its done today, in the service operators if we move it here, especially if you have multiple parameters in the struct and you want to allow the user to only specify the ones they want to change from the default. otherwise you'd always have to specify all.

Reply:
I suppose that makes sense. They will have to set value for at least one selector anyway.

Reply:
sorry, don't get your reply. can you explain it.